### PR TITLE
Refinements to minimal java.d.ts.

### PR DESCRIPTION
**Pull request by jimlloyd**
@mhfrantz Changes necessary for the upcoming ts-java PR.

### CODE REVIEW

**Comment by jimlloyd** [comment]
Merging pending review.

File: java/java.d.ts [hunk 1]
```diff
@@ -42,10 +43,16 @@ declare module Java {
...
+    newShort(val: number): any;
+    newLong(val: number): any;
+    newFloat(val: number): any;
+    newDouble(val: number): any;
+
     import(className: string): any;
     newInstance(className: string, ...args: any[]): void;
     newInstanceSync(className: string, ...args: any[]): any;
-    newInstancePromise(className: string, ...args: any[]): Promise<any>;
+    newInstanceP(className: string, ...args: any[]): Promise<any>;
```
Comment (mhfrantz) [inline java/java.d.ts]: How did this change from *Promise to *P?

**Comment by jimlloyd** [comment]
Node-java creates this method iff it is initialized with an `asyncOptions` object that defines the `promisify` function, and a `promiseSuffix`. It uses the supplied promise suffix. Formerly the ts-java BDD packages used different asyncOptions we have standardized on 'P' for the promise suffix.

The other two variants of `newInstance` are always created by node-java, and for arcane reasons it is not possible to defer creation of them until asyncOptions is specified, so they must use hard-coded names. This is documented in the node-java [README](https://github.com/joeferner/node-java#special-note-about-the-exported-module-functions-newinstance-callmethod-and-callstaticmethod).

It probably makes sense to modify node-java to use the hard-coded promise suffix "Promise" instead of the supplied suffix for `newInstance` (and the other two methods mentioned in the README) so that java.d.ts can be independent of `asyncOptions`.

**Comment by mhfrantz** [comment]
LGTM

**Pull request merged**